libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,333 @@
+import {
+  ClientProviderEvents,
+  EvaluationContext,
+  FlagMetadata,
+  FlagNotFoundError,
+  FlagValue,
+  GeneralError,
+  Hook,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderFatalError,
+  ResolutionDetails,
+  TypeMismatchError,
+} from '@openfeature/web-sdk';
+import { OfrepWebProviderOptions } from './model/ofrep-web-provider-options';
+import {
+  EvaluationFailureErrorCode,
+  EvaluationRequest,
+  EvaluationResponse,
+  handleEvaluationError,
+  isEvaluationFailureResponse,
+  isEvaluationSuccessResponse,
+  OFREPApi,
+  OFREPApiFetchError,
+  OFREPApiTooManyRequestsError,
+  OFREPApiUnauthorizedError,
+  OFREPForbiddenError,
+  RequestOptions,
+} from '@openfeature/ofrep-core';
+import {
+  InvalidContextError,
+  ParseError,
+  StandardResolutionReasons,
+  TargetingKeyMissingError,
+} from '@openfeature/core';
+import { isResolutionError, ResolutionError } from './model/resolution-error';
+import { BulkEvaluationStatus, EvaluateFlagsResponse } from './model/evaluate-flags-response';
+
+export class OFREPWebProvider implements Provider {
+  DEFAULT_POLL_INTERVAL = 30000;
+
+  readonly metadata = {
+    name: 'OpenFeature Remote Evaluation Protocol Web Provider',
+  };
+  readonly runsOn = 'client';
+  events = new OpenFeatureEventEmitter();
+  hooks?: Hook[] | undefined;
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+  // _options is the options used to configure the provider.
+  private _options: OfrepWebProviderOptions;
+
+  private _ofrepAPI: OFREPApi;
+  private _etag: string | null;
+  private _pollingInterval: number;
+  private _retryPollingAfter: Date | undefined;
+  private _cache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError } = {};
```

Comment:
Nit: Maybe we can have a cache type as it is used four time. But we do not have to.